src/hardware/vga_other.cpp

```diff
@@ -346,13 +346,18 @@ static void update_cga16_color(void) {
 			chroma_signals[j+1][i] = a + b*sin(x*tau) + c*cos(x*tau) + d*sin(x*2*tau);
 		}
 	}
-	Bitu CGApal[4] = {
+	// PC CGA palette table
+	const uint8_t cga_pal_fg_both_comp = color_sel || bw ? 1 : 0;
+	const uint8_t cga_pal_fg_color_comp = color_sel && !bw ? 1 : 0;
+	const uint8_t cga_pal_bg_comp = background_i ? 8 : 0;
+	const uint8_t CGApal[] = {
 		overscan,
-		static_cast<Bitu>(2 + (color_sel||bw ? 1 : 0) + (background_i ? 8 : 0)),
-		static_cast<Bitu>(4 + (color_sel&&!bw? 1 : 0) + (background_i ? 8 : 0)),
-		static_cast<Bitu>(6 + (color_sel||bw ? 1 : 0) + (background_i ? 8 : 0))
-	};
-	Bitu PCJRpal[4] = {vga.attr.palette[0], vga.attr.palette[1],
+	        static_cast<uint8_t>(2 + cga_pal_fg_both_comp + cga_pal_bg_comp),
+	        static_cast<uint8_t>(4 + cga_pal_fg_color_comp + cga_pal_bg_comp),
+	        static_cast<uint8_t>(6 + cga_pal_fg_both_comp + cga_pal_bg_comp)};
+
+	// PCjr CGA palette table
+	const uint8_t PCJRpal[] = {vga.attr.palette[0], vga.attr.palette[1],
```

Comment:
1. It seems to me like this whole calculation would make more sense if we were turning bits on and off instead of adding numbers… At the very least I suspect we should use hex instead of decimal integers - maybe using bitops instead of addition as well (not sure).
2. Reformat lines 349-361, please.

Reply:
Bitset approach would be idea if we could set individual bits at construction time - but the class isn't very flexible in this regard.  Here's what I came up with:

``` c++
	// PC CGA palette table
	std::bitset<8> CGApal[] = {overscan, 2, 4, 6};
	CGApal[1].set(0, color_sel || bw);
	CGApal[1].set(3, background_i);

	CGApal[2].set(0, color_sel && !bw);
	CGApal[2].set(3, background_i);

	CGApal[3].set(0, color_sel || bw);
	CGApal[3].set(3, background_i);
```

It's not that readable, and no longer a const-object. 

Also, getting values from the bitset requires `to_ulong()` and then casting as needed:

``` c++
rgbi = static_cast<uint8_t>(machine_palette[index].to_ulong());
```

Will try bit-wise operations next.

Reply:
Bit-wise is much better:

``` c++
	// PC CGA palette table
	const auto cga_base_comp = (color_sel || bw) | (background_i << 3);
	const auto cga_color_comp = (color_sel && !bw) | (background_i << 3);
	const int CGApal[] = {
	        overscan,
	        2 | cga_base_comp,
	        4 | cga_color_comp,
	        6 | cga_base_comp,
	};

	// PCjr CGA palette table
	const int PCJRpal[] = {
	        vga.attr.palette[0],
	        vga.attr.palette[1],
	        vga.attr.palette[2],
	        vga.attr.palette[3],
	};
```